Jenkinsfile

```diff
@@ -506,6 +506,15 @@ pass = ${OBS_CREDENTIALS_PASSWORD}
                                 )
                             }
                         }
+                        sh """
+                          set -e +x
+                          FILE=\$(ls scf-*linux-amd64*.zip)
+                          # Escape twice or the url will be unescaped when passed to the Jenkins form. It will then not work in the script.
+                          ENCODED_FILE_NAME=\$(ruby -e "require 'cgi'; puts CGI.escape('\${FILE}')")
+                          ENCODED_CAP_BUNDLE_URI=\$(ruby -e "require 'cgi'; puts CGI.escape('https://s3.amazonaws.com/${params.S3_BUCKET}/${params.S3_PREFIX}${distSubDir()}${distPrefix()}\${ENCODED_FILE_NAME}')")
+                          ENCODED_BUILD_URL=\$(ruby -e "require 'cgi'; puts CGI.escape('${BUILD_URL}')")
```

Comment:
This is intentionally manual because we don't want to open PRs against [the helm repo backend](https://github.com/suse/kubernetes-charts-suse-com) for every build. The release manager [is the one that decides](https://github.com/SUSE/cloudfoundry/wiki/CAP-Release-Process#sle) what charts should be released. I'm not sure if it makes sense to document this (business) logic here.

Regarding escaping, can we mix Groovy and sh? I mean, if we prepare the escaped string in Groovy how can we expose them to the sh script? We tried to do that but couldn't make it work.

Reply:
Given that it's in a `"""` block, yes; `${}` is actually more like `eval()` in groovy, I think.  Of course, since this block is so simple we could also port this to groovy instead (we grab the list of files where we do the S3 upload, so there's code from there to copy off of)